### ISSUE TITLE
Warning in UnwrappedLineFormatter.cpp while building llvm

### ISSUE THREAD

**Issue by anutosh491** [llvm/llvm-project#130923]
I am building llvm (clang/lld) against emscripten for a wasm usecase and i see this warning. (Not sure the same shows up while building for other platforms)

Not harmful and can still get my use case working but I thought of reporting it.

```
 │ │ $SRC_DIR/clang/lib/Format/UnwrappedLineFormatter.cpp:669:31: warning: comparison of integers of different signs: 'typename iterator_traits<AnnotatedLine *const *>::difference_type' (aka 'long') and 'const unsigned int' [-Wsign-compare]
 │ │   669 |       if (std::distance(I, E) <= N)
 │ │       |           ~~~~~~~~~~~~~~~~~~~ ^  ~
 │ │ 1 warning generated.
```

**Comment by owenca** [comment]
Were you building on a 32-bit system? What's `sizeof(long)` reported by emscripten?
